deploy/service.tf

```
@@ -0,0 +1,189 @@
+locals {
+  volumes = concat(var.docker_volumes, var.efs_volumes, var.fsx_volumes, var.bind_mount_volumes)
+  task_role_arn           = try(var.task_role_arn[0], tostring(var.task_role_arn), "")
+  enable_ecs_service_role = var.network_mode != "awsvpc" && length(var.ecs_load_balancers) >= 1
+}
+
+resource "aws_ecs_task_definition" "this" {
+  family                   = var.name
+  container_definitions    = var.container_definition_json
+  requires_compatibilities = [var.launch_type]
+  network_mode             = var.network_mode
+  cpu                      = var.task_cpu
+  memory                   = var.task_memory
+  execution_role_arn       = var.task_exec_role_arn
+  task_role_arn            = length(local.task_role_arn) > 0 ? local.task_role_arn : join("", aws_iam_role.ecs_task.*.arn)
```

```suggestion
  task_role_arn            = length(local.task_role_arn) > 0 ? local.task_role_arn : aws_iam_role.ecs_task.[0].arn
```
Any reason why this wouldn't do the job?